python/pyarrow/_compute_generated.py

```diff
@@ -0,0 +1,9381 @@
+        calendar unit.
+        For example: rounding to multiple of days since the beginning of the
+        month or to hours since the beginning of the day.
+        Exceptions: week and quarter are not used as greater units,
+        therefore days will be rounded to the beginning of the month not
+        week. Greater unit of week is a year.
+        Note that ceiling and rounding might change sorting order of an array
+        near greater unit change. For example rounding YYYY-mm-dd 23:00:00 to
+        5 hours will ceil and round to YYYY-mm-dd+1 01:00:00 and floor to
+        YYYY-mm-dd 20:00:00. On the other hand YYYY-mm-dd+1 00:00:00 will
+        ceil, round and floor to YYYY-mm-dd+1 00:00:00. This can break the
+        order of an already ordered array.
+    options : pyarrow.compute.RoundTemporalOptions, optional
+        Alternative way of passing options.
+    memory_pool : pyarrow.MemoryPool, optional
+        If not passed, will allocate memory from the default memory pool.
+    """
+
+    _computed_options = _handle_options(
+        'ceil_temporal',
+        pyarrow._compute.RoundTemporalOptions,
+        options,
+        (),
+        multiple=multiple,
+        unit=unit,
+        week_starts_monday=week_starts_monday,
+        ceil_is_strictly_greater=ceil_is_strictly_greater,
+        calendar_based_origin=calendar_based_origin
+    )
+    func = pyarrow._compute.get_function('ceil_temporal')
+
+    if isinstance(timestamps, Expression):
+        return Expression._call(
+            'ceil_temporal',
+            [timestamps],
+            _computed_options
+        )
+
+    return (
+        func.call([timestamps], _computed_options, memory_pool)
+    )
+
+
+def choose(indices, *values, memory_pool=None):
+    """Choose values from several arrays.
+
+    For each row, the value of the first argument is used as a 0-based index
+    into the list of `values` arrays (i.e. index 0 selects the first of the
+    `values` arrays). The output value is the corresponding value of the
+    selected argument.
+
+    If an index is null, the output will be null.
+
+    This wraps the "choose" compute function in the Arrow C++ library.
+
+    Parameters
+    ----------
+    indices : Array-like or scalar-like
+        Argument to compute function.
+    *values : Array-like or scalar-like
+        Argument to compute function.
+    memory_pool : pyarrow.MemoryPool, optional
+        If not passed, will allocate memory from the default memory pool.
+    """
+
+    func = pyarrow._compute.get_function('choose')
+    return (
+        func.call([indices, *values], memory_pool=memory_pool)
+    )
+
+
+def coalesce(*values, memory_pool=None):
+    """Select the first non-null value.
+
+    Each row of the output will be the value from the first corresponding
+        input
+    for which the value is not null. If all inputs are null in a row, the
+        output
+    will be null.
```

Comment:
That's probably from trying to keep the line length within flake8 max? We might need to "reflow" the text then ... (or find a way to indicate to flake8 to ignore this file)

Reply:
Ah - I see where this is coming from.  This is because the cpp-derived doc [for coalesce, as linked](https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/kernels/scalar_if_else.cc#L2739) has newlines which would be removed in the restructured text spec, but not before textwrap is being called. Let me work on this 

From the cpp source referenced:

```cpp
const FunctionDoc coalesce_doc{
    "Select the first non-null value",
    ("Each row of the output will be the value from the first corresponding input\n"
     "for which the value is not null. If all inputs are null in a row, the output\n"
     "will be null."),
    {"*values"}};
```